### DISCUSSION TITLE
ASUS ROG Strix G17 (G713RS-LL063WA) keyboard backlight issue

### DISCUSSION THREAD

**Discussion by mRx-z3d** [seerge/g-helper#2150] [Q&A] [answered]
Hi there,

I've been using g-helper for a while but can't seem to solve a problem I'm having. I have set the keyboard backlight to some nice rgb colours by using the 'rainbow' and then 'rain' to stop on the selected colours. This selection lasts between laptop restarts, but always returns to the default(?) red colour after power off/on. Does anyone know how to keep the selected rgb keyboard backlight settings? I have used different versions of g-helper, but this problem is always present. :\

[config.json](https://github.com/seerge/g-helper/files/14469519/config.json)
[log.txt](https://github.com/seerge/g-helper/files/14469520/log.txt)

**Comment by seerge** [comment]
@mRx-z3d hello, while you boot your keyboard may be static, but as soon you start G-Helper it sets last used mode (the one you see selected in dropdown)

Based on what I see it's Rain
```
02.03.2024 13:10:34: AuraMode: Rain
02.03.2024 13:10:34: USB 19B6: 5D-B3-00-05-FF-FF-FF-EB-00-00-FF-FF-FF-00-00-00-00
```

**Reply by mRx-z3d** [reply]
Hi @seerge, thans for looking into it.
Even though the log shows AuraMode: Rain, it changes to static red after power off/on. The funny thing is that it can maintain the Rain with rgb setting between reboots.
I did a simple test by setting the keyboard backlight to static, power off/on. Here it is able to keep the selected static colour, although you can see that the g-helper still shows 'Rain' as does the log and config file. I even set the config.json to read-only, where the aura_mode: 5, but the keyboard colour remains static :)

![IMG20240302165157](https://github.com/seerge/g-helper/assets/48696942/d8e13324-23e6-4e84-beff-dbdc0ffdd4be)

![IMG20240302165533](https://github.com/seerge/g-helper/assets/48696942/67f6b65e-7fdd-4e43-aa2d-c657bd78fe03)

So I'm no longer sure how it works. Is there a file other than config.json that holds the g-helper settings?

**Reply by seerge** [reply]
@mRx-z3d I'm don't really understand what you mean. 

On all your screenshots it says Rain (in logs and in app UI). Where did you set static ? There is only one ``config.json file``. Please keep in mind, that if you edit it manually - you need to QUIT app beforehand. 

As for what backlight your device has during power on / off (during windows boot), it's not controlled by app and depends on your specific hardware :) On most models it remebers last used mode of course. 

I also see that you have Asus services running in background, I would advice to **stop them** to be sure nothing interferes.

**Reply by mRx-z3d** [reply]
@seerge I've made a short video to help you understand the problem.
First, I set the keyboard backlight to 'Rainbow', turned my laptop off and then turned it on. You can see that the 'Rainbow' mode works. Then I change the mode to 'Star' and switch my laptop off and on again. This time keyboard does not preserve the 'Star' mode and the backlight colour changes to static red, even though you can see in g-helper that it is still set to 'Star'. Do you know why this happens?
[video link](https://drive.google.com/file/d/1GsR7orVnTOCIsb97wPSAiL7-4H0GbZC6/view?usp=sharing)

BTW. What Asus services are running in background? All Asus related services are turned off, expect the MultiAntenna Service but I guess this should not be a problem, right?
![IMG20240302194352](https://github.com/seerge/g-helper/assets/48696942/2e97139b-9725-48c0-9986-eefd7769b2ef)

**Reply by seerge** [reply]
@mRx-z3d from your video it looks like when you select Star nothing changes ? Then it simply means your keyboard don't support this mode (all modes except Ambient are hardware level)

When you reboot in unsupported mode - keyboard reverts to static.

**Reply by mRx-z3d** [reply]
@seerge it may be hard to see in the video, but the 'Rainbow' mode is dynamic and when I switch to 'Star' mode the colours are static. I'm not sure which modes my laptop supports, but I had no problem setting these RGB backlight modes using Armory Crate.

**Reply by seerge** [reply]
@mRx-z3d your device does not support Star, I assume you also didn't have that "Star" in armoury, right?
It has only 4-zone RGB, and those  devices have much less modes.

What happens when you set Star - your keyboard animation just hangs :) so it's static. You can test that by switching to Star from some other mode, and probably behavior would be different.

**Reply by mRx-z3d** [reply]
@seerge I think we are almost there :) Since my keyboard only has 4-zone RGB, is there a way to set a different colour for each zone with g-helper?

**Reply by seerge** [reply, marked as answer]
@mRx-z3d no, you can't customize colors individually per Zone. You can try OpenRGB if you want some very specific customization.

G-Helper supports only modes hardcoded into your keyboard firmware
